### ISSUE TITLE
Player is able to see through the block above them..

### ISSUE THREAD

**Issue by wwar-XeStro** [minetest/minetest#11319]
##### Minetest version

```
5.4,1
```

##### OS / Hardware

Operating system:
CPU: engine issue.. works on every OS..

GPU model:
OpenGL version:

##### Summary

a player can see through the block above them...

##### Steps to reproduce

When a player uses sneak and jump at the sametime under a block they float a bit: (i think this was already reported)

![Known](https://user-images.githubusercontent.com/72745616/120896610-05e94880-c62b-11eb-9186-84f6e8df0fe5.png)

Using 3rd person and aiming the camera to down (pitch: -75° to 89.5°)
Makes a player be able to see through that block above them.. (im standing under the lamp)

![Known2](https://user-images.githubusercontent.com/72745616/120896730-71331a80-c62b-11eb-9304-be8c76577575.png)

I think the reason is the hitbox of a player being shorter than needed and makes the eye offset colser than needed to a block causing it to by pass that block...

**Comment by Emojigit** [comment]
cannot reduce on 5.5.0-dev

**Comment by SmallJoker** [comment]
You could check whether #9211 fixes your issue, but due to the bad performance I think starting the node collision checks earlier would fix the problem too, though your view might end up being inside the player's head which does not look nice.

**Comment by wwar-XeStro** [comment]
> though your view might end up being inside the player's head which does not look nice.

something like these? 

![known2](https://user-images.githubusercontent.com/72745616/120920943-008f0b00-c6ca-11eb-9907-928883dfbec5.png)

![known](https://user-images.githubusercontent.com/72745616/120920970-169ccb80-c6ca-11eb-9c74-561ef29d09b4.png)

After Minetest 5.4 this also happens

![known3](https://user-images.githubusercontent.com/72745616/120921035-78f5cc00-c6ca-11eb-8a50-11237d6affa0.png)

(I dont use any shaders and didn't edit anything other than FOV (120) if that is related)

**Comment by SmallJoker** [comment]
Yes, especially the first one. The head node would somehow have to be set invisible where needed. The last image is caused by a single raytracing line, hence cannot take the edges of the visible areas into account.

**Comment by wwar-XeStro** [comment]
aa.. i didnt use the #9211 thing.. that happens all the time, and in the last imageim not near edges, i was standing in the red area: 

![known-](https://user-images.githubusercontent.com/72745616/120921624-45687100-c6cd-11eb-9a6e-ee3ac78f5ad2.png)

**Comment by wwar-XeStro** [comment]
This stuff only happen in singleplayer.

**Comment by appgurueu** [comment]
Since this issue is purely about the visual aspect, isn't this a duplicate #12429 of and should as such be closed by #15055?
